example/lib/example/socket_client.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Example.SocketClient do
+  @moduledoc false
+  require Logger
+  alias Phoenix.Channels.GenSocketClient
+  @behaviour GenSocketClient
+
+  def start_link() do
+    GenSocketClient.start_link(
+          __MODULE__,
+          Phoenix.Channels.GenSocketClient.Transport.WebsocketClient,
+          "ws://localhost:4000/socket/websocket"
+        )
+  end
+
+  def init(url) do
+    send(self(), :connect)
+    {:ok, url, %{first_join: true, ping_ref: 1}}
+  end
+
+  def handle_connected(transport, state) do
+    Logger.info("connected")
+    GenSocketClient.join(transport, "ping")
+    {:ok, state}
+  end
+
+  def handle_disconnected(reason, state) do
+    Logger.error("disconnected: #{inspect reason}")
+    Process.send_after(self(), :connect, :timer.seconds(1))
+    {:ok, state}
+  end
+
+  def handle_joined(topic, _payload, _transport, state) do
+    Logger.info("joined the topic #{topic}")
+
+    if state.first_join do
+      :timer.send_interval(:timer.seconds(1), self(), :ping_server)
+      {:ok, %{state | first_join: false, ping_ref: 1}}
+    else
+      {:ok, %{state | ping_ref: 1}}
```

Comment:
Why are you resetting the `ping_ref` upon subsequent joins? If I understand it correctly this would happen if we join additional topics? Shouldn't the ping ref in that case remain as it is?